### ISSUE TITLE
multi update

### ISSUE THREAD

**Issue by dvv** [marcello3d/node-mongolian#29]
`coll.updateAll({},{foo:1},function(){...})` reports success, but no changes are made. In real mongo shell, it would bark: "multi update only works with $ operators"

**update**: a light fix is to do `if (multi && !newObj.$set) newObj = {$set: newObj}` before composing  mongo command in `Collection#update`

--Vladimir

**Comment by marcello3d** [comment]
Thanks for the suggestion. I'll have to check more than $set, so maybe something like...

```
for (var x in newObj) if (x[0] != '$') throw new Error("multi update only works with $ operators or similar")
```

I'm curious how they do it in the mongo shell, since their code doesn't seem to mention it.

**Comment by dvv** [comment]
`x.charAt(0)`, i'd say

**Comment by marcello3d** [comment]
Is there a performance difference on v8 for [] vs charAt ?

**Comment by marcello3d** [comment]
Interesting, it looks like, if the index is < length, array notation is a bit faster on Chrome 13.0.782. If index > length, it's substantially slower: http://jsperf.com/array-notation-vs-charat

**Comment by marcello3d** [comment]
Apparently this is a server-side error, so we shouldn't be doing this check in the driver. http://www.google.com/codesearch#qPZGSFlTFyk/db/update.cpp&q=%22multi%20update%20only%20works%20with%20$%20operators%22&type=cs&l=1234

So now the question is why the error isn't being caught as an error!

**Comment by marcello3d** [comment]
Ok, I've tracked it down, and this will be resolved once I commit my fix for #15.

**Comment by dvv** [comment]
Great
